mobile/src/main/java/net/activitywatch/android/watcher/UsageStatsWatcher.kt

```diff
@@ -194,47 +233,82 @@ class UsageStatsWatcher constructor(val context: Context) {
             // TODO: Fix issues that occur when usage stats events are out of order (RESUME before PAUSED)
             var heartbeatsSent = 0
             val usageEvents = usm.queryEvents(lastUpdated?.toEpochMilli() ?: 0L, Long.MAX_VALUE)
+            var activeAppEvent : UsageEvents.Event?
+            activeAppEvent = null
+
             nextEvent@ while(usageEvents.hasNextEvent()) {
                 val event = UsageEvents.Event()
                 usageEvents.getNextEvent(event)
 
-                // Log screen unlock
                 if(event.eventType !in arrayListOf(UsageEvents.Event.ACTIVITY_RESUMED, UsageEvents.Event.ACTIVITY_PAUSED)) {
-                    if(event.eventType == UsageEvents.Event.KEYGUARD_HIDDEN){
-                        val timestamp = DateTimeUtils.toInstant(java.util.Date(event.timeStamp))
-                        // NOTE: getLastEventTime() returns the last time of an event from  the activity bucket(bucket_id)
-                        // Therefore, if an unlock happens after last event from main bucket, unlock event will get sent twice.
-                        // Fortunately not an issue because identical events will get merged together (see heartbeats)
-                        ri.heartbeatHelper(unlock_bucket_id, timestamp, 0.0, JSONObject(), 0.0)
-                    }
-                    // Not sure which events are triggered here, so we use a (probably safe) fallback
-                    //Log.d(TAG, "Rare eventType: ${event.eventType}, skipping")
                     continue@nextEvent
                 }
 
-                // Log activity
-                val awEvent = Event.fromUsageEvent(event, context, includeClassname = true)
-                val pulsetime: Double
+                // do not include launchers, since they are used all the time to switch between apps. It distorts the timeline while
+                // it is more part of the OS than an app
+                if( event.packageName.contains("launcher", false) ) {
+                    Log.d(TAG,"Skipping launcher event for package " + event.packageName)
+                    continue@nextEvent
+                }
```

Comment:
I think it can be tracked but I'm not sure the info will be very valuable. Most other time trackers I know of seem to ignore launchers as well. It's like tracking the time you spent just looking at the desktop in Windows. 

But yes, this code can be skipped and it *should* still work.

Reply:
I had similar thoughts, but we think we'll just let the user collect as much info as possible. They can later filter it as they please.